File: Client/src/types/listing.ts
// Enums matching backend (using const objects due to erasableSyntaxOnly)
export const ListingType = {
  Selling: 1,
  Buying: 2
} as const

export type ListingType = typeof ListingType[keyof typeof ListingType]

export const ItemCondition = {
  New: 1,
  NotUsed: 2,
  Used: 3,
  Damaged: 4
} as const

export type ItemCondition = typeof ItemCondition[keyof typeof ItemCondition]

export const Currency = {
  EUR: 1,
  USD: 2,
  RSD: 3,
  BAM: 4,
  HRK: 5
} as const

export type Currency = typeof Currency[keyof typeof Currency]

export const ListingStatus = {
  Active: 1,
  Sold: 2,
  Inactive: 3,
  Expired: 4,
  Pending: 5,
  Rejected: 6
} as const

export type ListingStatus = typeof ListingStatus[keyof typeof ListingStatus]

// API Request/Response types
export interface CreateListingRequest {
  caption: string
  description: string
  categoryId: number
  listingType: ListingType
  price: number
  currency: Currency
  isPriceFixed: boolean
  acceptsTrade: boolean
  itemCondition: ItemCondition
  availableImmediately: boolean
  deliveryAvailable: boolean
  inPersonPickup: boolean
  city: string
  contactName: string
}

export interface CreateListingResponse {
  id: number
  caption: string
  message: string
}

// Form data type
export interface ListingFormData {
  caption: string
  description: string
  category: string
  listingType: 'selling' | 'buying'
  price: string
  currency: string
  isPriceFixed: boolean
  acceptsTrade: boolean
  itemCondition: '' | 'new' | 'not-used' | 'used' | 'damaged'
  availableImmediately: boolean
  deliveryAvailable: boolean
  inPersonPickup: boolean
  city: string
  contactName: string
}

File: Server/Oglasnik/Oglasnik.Data/Entities/Listing.cs
using System.ComponentModel.DataAnnotations;
using Oglasnik.Contracts.Enums;

namespace Oglasnik.Data.Entities;

public class Listing : Base
{
    // Basic Information
    [Required]
    [MaxLength(100)]
    public string Caption { get; set; } = string.Empty;

    [Required]
    [MaxLength(2000)]
    public string Description { get; set; } = string.Empty;

    // Category Reference
    public long CategoryId { get; set; }

    // Listing Type & Price
    [Required]
    public ListingType ListingType { get; set; } = ListingType.Selling;

    [Required]
    public decimal Price { get; set; }

    [Required]
    public Currency Currency { get; set; } = Currency.EUR;

    public bool IsPriceFixed { get; set; } = true;

    public bool AcceptsTrade { get; set; } = false;

    // Item Condition
    [Required]
    public ItemCondition ItemCondition { get; set; }

    // Availability
    public bool AvailableImmediately { get; set; } = true;

    // Delivery Options
    public bool DeliveryAvailable { get; set; } = false;

    public bool InPersonPickup { get; set; } = true;

    // Location & Contact
    [Required]
    [MaxLength(100)]
    public string City { get; set; } = string.Empty;

    [Required]
    [MaxLength(100)]
    public string ContactName { get; set; } = string.Empty;

    // User Reference
    public long UserId { get; set; }

    // Listing Status
    [Required]
    public ListingStatus Status { get; set; } = ListingStatus.Active;

    // Engagement Metrics
    public int ViewCount { get; set; } = 0;

    public int FollowCount { get; set; } = 0;

    // Expiration (optional - for featured/promoted listings)
    public DateTime? ExpiresOnUtc { get; set; }

    // Navigation Properties
    public UserAccount User { get; set; } = null!;

    public Category Category { get; set; } = null!;

    public ICollection<ListingImage> Images { get; set; } = new List<ListingImage>();
}
